ga4gh/frontend.py

```diff
@@ -391,6 +468,57 @@ def searchDatasets(version):
         version, flask.request, app.backend.searchDatasets)
 
 
+@app.route('/oauth2callback', methods=['GET'])
+def oidcCallback():
+    """
+    Once the authorization provider has cleared the user, the browser
+    is returned here with a code. This function takes that code and
+    checks it with the authorization provider to prove that it is valid,
+    and get a bit more information about the user (which we don't use).
+
+    A token is generated and given to the user, and the authorization info
+    retrieved above is stored against this token. Later, when a client
+    connects with this token, it is assumed to be a valid user.
+
+    :return: A display of the authentication token to use in the client.
+    """
+    if app.oidcClient is None:
+        return 'Authentication not enabled.', 404
```

Comment:
We're using exceptions everywhere else to signal a HTTP error though. It would be better to use the same mechanism, so we can manage the content types, JSON encoding, etc.